### PR TITLE
Refactor vocab module, rename subword vocabs.

### PR DESCRIPTION
**Pull request by sebpuetz**
FinalfusionSubwordVocab -> BucketSubwordVocab,
FinalfusionNGramVocab -> ExplicitSubwordVocab, NGramIndexer ->
ExplicitIndexer. Move SimpleVocab, SubwordVocab, VocabWrap to
their own modules.

### CODE REVIEW

**Comment by sebpuetz** [comment]
I amended this commit to remove the `Clone` bound on `Vocab`. We don't seem to rely on it anywhere and it's possible to add the `Clone` bound explicitly whenever necessary.

File: src/chunks/vocab/mod.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+mod subword;
+pub use subword::{
+    BucketSubwordVocab, ExplicitSubwordVocab, FastTextSubwordVocab, NGramIndices, SubwordIndices,
+    SubwordVocab,
+};
+
+mod simple;
+pub use simple::SimpleVocab;
+
+mod wrapper;
```
Comment (danieldk) [inline src/chunks/vocab/mod.rs]: We call the module `wrappers` for storage, so we should do the same here.

Comment (sebpuetz) [reply src/chunks/vocab/mod.rs]: There it's actually multiple wrappers, `StorageWrap` and `StorageViewWrap`. If you really don't like it, I can change it

**Comment by sebpuetz** [comment]
Did the renaming, also caught another case of unnecessary `Clone` bounds.

File: src/chunks/vocab/mod.rs (hunk 1 above)
Comment (danieldk) [reply src/chunks/vocab/mod.rs]: Yep, naming consistency between the modules is nice.

**Pull request merged**